qt/Utils.cc

```diff
@@ -301,3 +302,31 @@ QColor Utils::getFadedColor(QColor const& color)
     faded_color.setAlpha(128);
     return faded_color;
 }
+
+QString Utils::getDynamicDownloadDir(Prefs const& prefs, QString const& torrentName)
+{
+    if (prefs.getBool(Prefs::DOWNLOAD_DIR_DYNAMIC_ENABLED))
+    {
+        auto const filter_table = prefs.getString(Prefs::DOWNLOAD_DIR_DYNAMIC_TABLE);
+        auto const rows = filter_table.split(QStringLiteral(";"));
```

Comment:
`;` and `,` are maybe not the most common characters to show up in a regex, but this still smells brittle in that we're tokenizing the string based on user input s.t. the user mistyping their regex could cause very strange behavior here. :smiley: 

Maybe if we stored this internally not as a string, but as an array of <mode, regex, path> where mode is an enum of name or announce_url? That way we wouldn't have to manually parse every time. Also note that the variants we pass to libtransmission don't _have_ to be strings, they could be a vector of these tuples. This would get percolated down to `settings.json` holding the list instead of simple key/value string pairs and should Just Work